revise Stratum 0 part (#6)
cvmfs-contrib/cvmfs-tutorial-2021: docs/02_stratum0_client.md

For now, we are going to set up a Stratum 0, create a repository, and access it by connecting from a client machine directly to the Stratum 0 server.

!!! warning
    **Directly connecting to the Stratum 0 is** ***not*** **recommended**, but it is a good way to get started.

    In the [next hands-on part of this tutorial](03_stratum1_proxies.md) we will remedy this by setting up a Stratum 1
    replica server and squid proxy, and accessing the repository that way instead.

## 2.1 Setting up the Stratum 0

### 2.1.1 Requirements

#### Resources

Due to the scalable design of CernVM-FS, the Stratum 0 server does not need a lot of resources in terms of CPU cores and memory;
just a few cores and a couple gigabytes of memory is sufficient.

Besides this, you do need sufficient storage space to store the contents of your repository. By default, CernVM-FS uses `/var/spool` as scratch space
while adding new files to the repository, and `/srv/cvmfs` as central repository storage location, but these locations can be modified in the
CernVM-FS server configuration settings later on.

#### Operating system

Several (popular) Linux distributions are supported by CernVM-FS,

> see [these requirements](https://github.com/cvmfs-contrib/cvmfs-tutorial-2021/wiki/Notes#stratum-1--proxies) for a full list.

Reviewer: Don't know how this url ended up here (it was already like that in my version, apparently), but it somehow points to the wiki of this github repo.
```suggestion
see [the Getting Started page of the CernVM-FS documentation](https://cvmfs.readthedocs.io/en/stable/cpt-quickstart.html#getting-the-software) for a full list.
```

[https://cvmfs.readthedocs.io/en/stable/cpt-repo.html#master-keys](https://cvmfs.readthedocs.io/en/stable/cpt-repo.html#master-keys).

### 2.1.6 Adding files to the repository

A new repository automatically gets a file named `new_repository` in its root (`/cvmfs/repo.organization.tld`). You can add more files by starting and publishing a transaction, which will be explained in more detail in the [3rd hands-on part of this tutorial](04_publishing.md).

For now it is enough to just run the following commands to add a simple `hello.sh` script to your repository.

First, start the transaction via `cvmfs_server transaction <name_of_repo>`:

```bash
# Change this to your repository/domain name!
MY_REPO_NAME=repo.organization.tld
```

```bash
sudo cvmfs_server transaction ${MY_REPO_NAME}
```

Next, add the file to the repository (in `/cvmfs/${MY_REPO_NAME}`).
If you made the current user the owner of the repository, you can do this without `sudo` since you have write
permissions to your repository:

```bash
echo '#!/bin/bash' > /cvmfs/${MY_REPO_NAME}/hello.sh
echo 'echo hello' >> /cvmfs/${MY_REPO_NAME}/hello.sh
chmod a+x /cvmfs/${MY_REPO_NAME}/hello.sh
```

> Complete the transaction by publishing the changes using `cvmfs_server publish <name_of_repo>`:

Reviewer: I think this last part of the sentence can be removed, otherwise the same thing is mentioned twice.
```suggestion
Complete the transaction by publishing the changes using:
```

Each CernVM-FS repository has a whitelist containing fingerprints of certificates that are allowed to sign the repository.
This whitelist has an expiration time of (by default) 30 days, so you regularly have to resign the whitelist.

There are several ways to do this, see for instance [the page about master keys](https://cvmfs.readthedocs.io/en/stable/cpt-repo.html#sct-master-keys)
in the CernVM-FS documentation.

If you keep the master key on our Stratum 0 sever, you can set up a simple cronjob for resigning the whitelist. For instance, make a file `/etc/cron.d/cvmfs_resign` with the following content to do this every Monday at 11:00:
```
0 11 * * 1 root /usr/bin/cvmfs_server resign repo.organization.tld
```

For the sake of this tutorial this is not really necessary of course, but it's an important aspect to be aware of.

### 2.1.8 Removing a repository

An existing repository can be removed easily by running:
```
sudo cvmfs_server rmfs repo.organization.tld
```

Obviously you should only do this when you actually *want* to get rid of the repository...

## 2.2 Setting up a client

Accessing a CernVM-FS repository on a client system involves three steps:

> 1. installing the CernVM-FS client package;
> - adding some configuration files for the repository you want to connect to;
> - running the CernVM-FS setup procedure that will mount the repository.

Reviewer: ```suggestion
1. installing the CernVM-FS client package;
2. adding some configuration files for the repository you want to connect to;
3. running the CernVM-FS setup procedure that will mount the repository.
```

Author: This is a trick to not having the number them ourselves (it looks fine in the rendered version with 1, 2, 3), but it looks confusing in the source file, so makes sense to change.

*Don't forget to rename or change these files according to the repository name and domain you are using!*

#### Public key of repository

`/etc/cvmfs/keys/organization.tld/repo.organization.tld.pub`

This file contains the public key of the repository you want to access.

You can copy this file from your Stratum 0 server, where it is stored under `/etc/cvmfs/keys/`.

#### Main configuration for repository

`/etc/cvmfs/config.d/repo.organization.tld.conf`

This file contains the main configuration for the repository you want to access, which should minimally contain the URL(s) of the Stratum 1 servers and the location of the public key.

We do not have a Stratum 1 server yet (we'll set that up in [the next hands-on part of this tutorial](03_stratum1_proxies.md)),
so we are going to connect directly to our Stratum 0 server instead. **You should not do this in production!**

A typical, minimal configuration should look as follows:  

```
CVMFS_SERVER_URL="http://<IP>/cvmfs/@fqrn@"
CVMFS_PUBLIC_KEY="/etc/cvmfs/keys/organization.tld/repo.organization.tld.pub"
```

***Replace the ``<IP>`` part with the IP address of your Stratum 0 server!***

> Note that the `CVMFS_SERVER_URL` should include the part `/cvmfs/@fqrn` exactly like that;

Reviewer: ```suggestion
Note that the `CVMFS_SERVER_URL` should include the part `/cvmfs/@fqrn@` exactly like that;
```
